### PR TITLE
Create Room screen

### PR DESCRIPTION
**Pull request by bmarty**
This PR also introduce:
- 2 form items for Epoxy
- a navigation view model to navigate between fragment inside an Activity

### CODE REVIEW

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/create/CreateRoomTask.kt [hunk 1]
```diff
@@ -54,6 +54,7 @@ internal class DefaultCreateRoomTask(private val roomAPI: RoomAPI,
             handlerThread.start()
             val handler = Handler(handlerThread.looper)
 
+            // TODO Maybe do the same code for join room request ?
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/create/CreateRoomTask.kt]: Maybe use RealmQueryLatch ? Should be tested first

File: vector/src/main/java/im/vector/riotredesign/features/form/FormEditTextItem.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import com.airbnb.epoxy.EpoxyModelClass
+import com.google.android.material.textfield.TextInputEditText
+import com.google.android.material.textfield.TextInputLayout
+import im.vector.riotredesign.R
+import im.vector.riotredesign.core.epoxy.VectorEpoxyHolder
+import im.vector.riotredesign.core.epoxy.VectorEpoxyModel
+import im.vector.riotredesign.core.platform.SimpleTextWatcher
+
+@EpoxyModelClass(layout = R.layout.item_form_text_input)
+abstract class FormEditTextItem : VectorEpoxyModel<FormEditTextItem.Holder>() {
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotredesign/features/form/FormEditTextItem.kt]: Yes, good idea to use epoxy for form!

File: vector/src/main/java/im/vector/riotredesign/features/home/HomeDetailFragment.kt [hunk 3]
```diff
@@ -89,7 +89,7 @@ class HomeDetailFragment : VectorBaseFragment() {
                 groupToolbarAvatarImageView
         )
         groupToolbarAvatarImageView.setOnClickListener {
-            vectorBaseActivity.notImplemented("Group click in toolbar")
+            (vectorBaseActivity as? HomeActivity)?.openDrawer()
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotredesign/features/home/HomeDetailFragment.kt]: Maybe hide this with an interface?

File: vector/src/main/java/im/vector/riotredesign/features/roomdirectory/RoomDirectoryActivity.kt [hunk 4]
```diff
@@ -17,21 +17,45 @@
...
+    private val navigationViewModel: RoomDirectoryNavigationViewModel by viewModel()
 
     override fun getLayoutRes() = R.layout.activity_simple
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         bindScope(getOrCreateScope(RoomDirectoryModule.ROOM_DIRECTORY_SCOPE))
+
+        navigationViewModel.navigateTo.observe(this, Observer { liveEvent ->
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotredesign/features/roomdirectory/RoomDirectoryActivity.kt]: Use observeEvent extension method

File: vector/src/main/java/im/vector/riotredesign/core/mvrx/NavigationViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.airbnb.mvrx.MvRxState
+import im.vector.riotredesign.core.platform.VectorViewModel
+import im.vector.riotredesign.core.utils.LiveEvent
+
+// MvRx require a state with at least one attribute
+data class NavigationState(val dummy: Boolean = false) : MvRxState
+
+abstract class NavigationViewModel<NavigationClass>(initialState: NavigationState) : VectorViewModel<NavigationState>(initialState) {
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotredesign/core/mvrx/NavigationViewModel.kt]: Maybe switch to a simple ViewModel as we don't need state?

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/create/CreateRoomTask.kt (hunk 1 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/create/CreateRoomTask.kt]: RealmQueryLatch is not used for the moment, why?

File: vector/src/main/java/im/vector/riotredesign/features/form/FormEditTextItem.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/riotredesign/features/form/FormEditTextItem.kt]: There are still side effect (ex: lost of focus when rotating screen), but for reusability, it's great

File: vector/src/main/java/im/vector/riotredesign/features/home/HomeDetailFragment.kt (hunk 3 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/riotredesign/features/home/HomeDetailFragment.kt]: I will use the navigationViewModel

**Pull request merged**